### PR TITLE
chore: Test shared user org select #4667

### LINKED ISSUE #4667: Add RTL tests to `@shared/UserOrgSelect.jsx`
Add a test to `@shared/UserOrgSelect.jsx` using the RTL framework.

## Acceptance Criteria

- [ ] Colocate test file to `@shared/UserOrgSelect.test.jsx`
- [ ] Test the label is showing as expected
- [ ] Test the error message shows when the select has been touched and has an error
- [ ] Test the options are available
- [ ] Test the default option `Please select an organization` on initial render
- [ ] Test the select change works
- [ ] Remove components `defaultProps` and them to the function args

### PR DESCRIPTION
**Pull request by apburnes**
Closes #4667

## Changes proposed in this pull request:
- Remove's default props and adds defaults to function args
- Fixes `htmlFor` to use the `id` prop
- Adds tests for select

## security considerations
None

### CODE REVIEW

**Comment by cloud-gov-pages-operations** [comment]
## :robot: This is an automated code coverage report
  Total coverage (lines): **14.83%**
  Coverage diff: **0.27%** :chart_with_upwards_trend:

File: frontend/shared/UserOrgSelect.jsx [hunk 1]
```diff
@@ -14,17 +14,17 @@ function makeOptions(opts) {
 const UserOrgSelect = ({
   className,
```
Comment (drewbo) [inline frontend/shared/UserOrgSelect.jsx:15]: I'd initialize `className` as `''` since it's printed

File: frontend/shared/UserOrgSelect.test.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const initialProps = {
+  className: '',
+  id,
+  label,
+  touched: false,
+  error: null,
+  mustChooseOption: false,
+  name,
+  onChange: spy(),
+  orgData: [
```
Comment (drewbo) [inline frontend/shared/UserOrgSelect.test.jsx:24]: Even though this one is pretty small, I'd put this in another file so we have some organization fixture data to work with

**Review by drewbo — CHANGES_REQUESTED** [review]
Two small items but looks good

**Review by drewbo — APPROVED** [review]

**Pull request merged**